### ISSUE TITLE
startup race condition / suggestion to address in salts_util.py

### ISSUE THREAD

**Issue by mgonzales71** [teverz/teverz#1]
I know MB is busy so I thought maybe you could consider this at some future update, etc...
(Thanks)

I've noticed Kodi will often hang on startup when add on fire up at startup.  In Kodi 18 less often but previous Kodi builds have only so many threads to work with when it comes to add on / scripts.

One thing that I have found to mitigate this is to update the following in the Various
salts_util.py in the various addons (DSRDplus, etc) (delay startup)

```
# Run a task on startup. Settings and mode values must match task name
def do_startup_task(task):
    run_on_startup = kodi.get_setting('auto-%s' % task) == 'true' and kodi.get_setting('%s-during-startup' % task) == 'true'
    if run_on_startup and not xbmc.abortRequested:
        logger.log('Service: Delaying startup task [%s] 45 seconds' % (task), log_utils.LOGWARNING)
        xbmc.sleep(45000)
        logger.log('Service: Running startup task [%s]' % (task), log_utils.LOGNOTICE)
        now = datetime.datetime.now()
        xbmc.executebuiltin('RunPlugin(plugin://%s/?mode=%s)' % (kodi.get_id(), task))
        db_connection.set_setting('%s-last_run' % (task), now.strftime("%Y-%m-%d %H:%M:%S.%f"))


```

Basically the only change is the addition of:
#
        logger.log('Service: Delaying startup task [%s] 45 seconds' % (task), log_utils.LOGWARNING)
        xbmc.sleep(45000)

Suppose you could go so far as to add a setting (Boolean) and then (number) for delay in seconds if enabled as well.

Thanks for your work and contributions!
